cmd/kubeadm/app/cmd/phases/init/addons_test.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package phases
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/kubernetes/cmd/kubeadm/app/cmd/options"
+)
+
+func TestGetAddonPhaseFlags(t *testing.T) {
+	tests := []struct {
+		casename string
+		name     string
+		want     []string
+	}{
+		{
+			casename: "test name is all",
+			name:     "all",
+			want: []string{options.CfgPath,
+				options.KubeconfigPath,
+				options.KubernetesVersion,
+				options.ImageRepository,
+				options.DryRun,
+				options.APIServerAdvertiseAddress,
+				options.ControlPlaneEndpoint,
+				options.APIServerBindPort,
+				options.NetworkingPodSubnet,
+				options.FeatureGatesString,
+				options.NetworkingDNSDomain,
+				options.NetworkingServiceSubnet,
+			},
+		}, {
+			casename: "test name is kube-proxy",
+			name:     "kube-proxy",
+			want: []string{options.CfgPath,
+				options.KubeconfigPath,
+				options.KubernetesVersion,
+				options.ImageRepository,
+				options.DryRun,
+				options.APIServerAdvertiseAddress,
+				options.ControlPlaneEndpoint,
+				options.APIServerBindPort,
+				options.NetworkingPodSubnet,
+			},
+		}, {
+			casename: "test name is coredns",
+			name:     "coredns",
+			want: []string{options.CfgPath,
+				options.KubeconfigPath,
+				options.KubernetesVersion,
+				options.ImageRepository,
+				options.DryRun,
+				options.FeatureGatesString,
+				options.NetworkingDNSDomain,
+				options.NetworkingServiceSubnet,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := getAddonPhaseFlags(tt.name)
```

Comment:
this just covered the func of `getAddonPhaseFlags` which is not an exported func, ideally, we want to have testcase to cover the exported func, e.g. `NewAddonPhase` func instead.

Can you pls create another one for that? here is the example you can follow, 
https://github.com/kubernetes/kubernetes/blob/5c72df728125530f15a519dacc9c308084dc586b/cmd/kubeadm/app/cmd/phases/init/certs_test.go#L42